engine/src/main/scala/cromwell/engine/workflow/workflowstore/SqlWorkflowStore.scala

```diff
@@ -30,18 +52,33 @@ case class SqlWorkflowStore(sqlDatabase: WorkflowStoreSqlDatabase) extends Workf
     *  startup initialization hook. */
   override def initialize(implicit ec: ExecutionContext): Future[Unit] = Future.successful(())
 
-  override def aborting(id: WorkflowId)(implicit ec: ExecutionContext): Future[Option[Boolean]] = {
-    sqlDatabase.setToAborting(id.toString)
+  override def aborting(id: WorkflowId)(implicit ec: ExecutionContext): Future[WorkflowStoreAbortResponse] = {
+    sqlDatabase.deleteOrUpdateWorkflowToState(
+      id.toString,
+      WorkflowStoreState.OnHold.toString,
+      WorkflowStoreState.Aborting.toString
+    ) map {
+      case (deleteOnHoldCount, _) if deleteOnHoldCount > 0 =>
+        WorkflowStoreAbortResponse.AbortedOnHold
+      case (_, None) =>
+        WorkflowStoreAbortResponse.NotFound
+      case (_, Some(true)) =>
+        WorkflowStoreAbortResponse.AbortingHeartbeatTimestampIsEmpty
+      case (_, Some(false)) =>
+        WorkflowStoreAbortResponse.AbortingHeartbeatTimestampNonEmpty
```

Comment:
Perhaps an ADT rather than this tuple type coming back from `deleteOrUpdateWorkflowToState`?

Reply:
Thought about the ADT, but couldn't come up with a clean SQL-like abstraction for the `sql` layer. (I try to keep the `sql` layer as close to stored-procedures as possible, keeping all non-db-logic in the services layer.) Here, in the middle/services layer, the multiple out parameters are immediately converted into an enum.